backend_app/api/utils/Persistence.js

```diff
@@ -0,0 +1,64 @@
+const { v4: uuidv4 } = require("uuid");
+const bcrypt = require("bcrypt");
+const { invokeAndSafelyClose } = require("../../services/MongoService");
+
+/*
+  -- USERS --
+*/
+
+/**
+ * Util to check if the user already exists in the database or not
```

Comment:
Si le but c'est de checker si un user existe j'appelerais ça checkIfUserExist et je passerait le user. Éventuellement il va pt avoir plusieurs façon de s'inscrire (genre avec numéro mobile) et le checkifuserexist pourrait appeler plusieurs méthodes du genre checkIfEmailExists et checkIfPhoneNumberExists

Reply:
Ah et quand ça retourne un boolean moi je trouve ça cool de nommer la fonction avec un is ou has genre isExistingUser

Reply:
pk on passerait le user. Au final le seule champ qui est garanti d'etre unique cest le email

Reply:
Ouin mon point c'est que éventuellement ce sera plus vrai. Ça va peut-être pouvoir être un numéro de cellulaire (éventuellement je pencherais plus vers le cell pour s'assurer que les users sont unique) mon point c'est que ça coute pas grand chose de le changer live alors que plus tard quand la fonction va être utilisée à plusieurs endroit ça va être tannant. Je garderais la fonction, mais j'en ajouterais une autre isExistingUser auquel on passe le user et qui callerait celle là et j'utiliserais isExistingUser dans la verification à la place

Reply:
L'affaire que j'aime moins avec le phone cest qu'on aurait abosulement besoin que le email soit valide parce que on va vouloir communiquer avec les usager genre s'ils ont eu un view, ou genre une demande. Pour ca je vois mal comment on pourrait le faire par texto, messemble que cest moins user friendly